prExplorer/client-app/src/features/products/dashboard/ProductDashboard.tsx

```diff
@@ -1,39 +1,27 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { Grid, GridColumn } from "semantic-ui-react";
-import { Product } from "../../../app/models/product";
 import ProductList from "./ProductList";
-import ProductDetails from "../details/ProductDetails";
-import ProductForm from "../form/ProductForm";
+import { useStore } from "../../../app/stores/store";
+import { observer } from "mobx-react-lite";
+import LoadingComponent from "../../../app/layout/LoadingComponent";
 
-interface Props {
-  products: Product[];
-  selectedProduct: Product | undefined;
-  selectProduct: (id: string) => void;
-  cancelSelectedProduct: () => void;
-  editMode: boolean;
-  openForm: (id: string) => void;
-  closeForm: () => void;
-  createOrEdit: (product: Product) => void;
-  deleteProduct: (id: string) => void;
-  submitting: boolean;
-}
 
-export default function ProductDashboard({ products, selectProduct, selectedProduct, cancelSelectedProduct, editMode, openForm, closeForm, createOrEdit, deleteProduct, submitting}: Props) {
+export default observer(function ProductDashboard() {
+
+  const {productStore} = useStore();
+  const {loadProducts, productRegistry} = productStore;
+
+  useEffect(() => {
+    if (productRegistry.size <= 1) loadProducts();
```

Comment:
It would be better if you use `productStore` directly in that case, i.e.:

```suggestion
    if (productRegistry.size <= 1) productStore.loadProducts();
```

currently, when I see this code, I don't know straightaway that `loadProducts()` is fom the store, which is quite an important information